### PR TITLE
Work on cli tools

### PR DESCRIPTION
**Pull request by yakutovicha**
In this PR I revised our CLI tools to further implement smoke testing.

### CODE REVIEW

**Comment by yakutovicha** [comment]
@NikoOinonen, @ProkopHapala, I quickly tested things, but might be I missed something. Please have a loook.

File: ppafm/cli/generateElFF.py [hunk 1]
```diff
@@ -1,178 +1,184 @@
...
+
+    electrostatic_potential *= -1  # Unit conversion, energy to potential (eV -> V)
+
+    # To fix.
+    # if common.params["tip"] == ".py":
+    #     # import tip
+    #     exec(compile(open("tip.py", "rb").read(), "tip.py", "exec"))
+    #     print(tipMultipole)
+    #     common.params["tip"] = tipMultipole
+    #     print("params['tip'] ", common.params["tip"])
```
Comment (NikoOinonen) [inline ppafm/cli/generateElFF.py:71]: What is this actually? Looks kinda hacky. Does removing this break any functionality? @ProkopHapala This is your code, I think.

File: ppafm/cli/relaxed_scan.py [hunk 2]
```diff
@@ -1,37 +1,39 @@
...
+    return new_fx, new_fy
 
-def main():
 
-    parser = PPU.CLIParser(
-        description='Perform a scan, relaxing the probe particle in a precalculated force field. '
-            'The generated force field is saved to Q{charge}K{klat}/OutFz.xsf.'
+def main():
+    parser = common.CLIParser(
+        description="Perform a scan, relaxing the probe particle in a precalculated force field. " "The generated force field is saved to Q{charge}K{klat}/OutFz.xsf."
```
Comment (NikoOinonen) [inline ppafm/cli/relaxed_scan.py:33]: Can be combined into a single string.

File: ppafm/cli/relaxed_scan.py [hunk 3]
```diff
@@ -46,146 +48,152 @@ def main():
...
+        if opt_dict["disp"]:
+            io.save_vec_field(dirname + "/PPdisp", pp_displacements, **data_info)
+
+        if opt_dict["pos"]:
+            io.save_vec_field(dirname + "/PPpos", pp_positions, **data_info)
+
+        if args.bI:
+            print("Calculating current from tip to the Boltzmann particle:")
+            i_in, lvec, _, atomic_info_or_head = io.load_scal_field("I_boltzmann", data_format=args.output_format)
+            i_out = interpolate_cartesian(i_in, pp_positions, cell=lvec[1:, :], result=None)
```
Comment (NikoOinonen) [inline ppafm/cli/relaxed_scan.py:194]: The meaning of `i_in` and `i_out` are unclear here. With the previous capital `I`, I would assume that these refer to some currents. The lower-case `i` could be anything.

File: ppafm/cli/generateDFTD3.py [hunk 4]
```diff
@@ -2,51 +2,57 @@
...
+from ..HighLevel import computeDFTD3
 
-def main():
 
-    parser = PPU.CLIParser(
-        description='Generate Grimme DFT-D3 vdW force field using the Becke-Johnson damping function. '
-            'The generated force field is saved to FFvdW_{x,y,z}.[ext].'
+def main():
+    parser = common.CLIParser(
+        description="Generate Grimme DFT-D3 vdW force field using the Becke-Johnson damping function. " "The generated force field is saved to FFvdW_{x,y,z}.[ext]."
```
Comment (NikoOinonen) [inline ppafm/cli/generateDFTD3.py:13]: Can be combined into a single string.

**Review by NikoOinonen — CHANGES_REQUESTED** [review]
Just a few minor things that I noticed.

File: ppafm/cli/plot_results.py [hunk 5]
```diff
@@ -1,28 +1,29 @@
...
+from ..HighLevel import symGauss
 
 mpl.use("Agg")
 print("plot WITHOUT Xserver")
 # this makes it run without Xserver (e.g. on supercomputer) # see http://stackoverflow.com/questions/4931376/generating-matplotlib-graphs-without-a-running-x-server
 
 
+HBAR = 6.58211951440e-16  # [eV.s]
+AUMASS = 1.66053904020e-27  # [kg]
+EVA2_TO_NM = 16.0217662  # [eV/A^2] / [N/m]
```
[the remark this answers is not in the document]
Comment (yakutovicha) [reply ppafm/cli/plot_results.py]: done in https://github.com/Probe-Particle/ppafm/pull/215/commits/2a649fd9d6632be5d9494d6528ee2ca121ef8a92

File: ppafm/cli/generateElFF.py [hunk 6]
```diff
@@ -1,178 +1,184 @@
...
-        if(args.tip_dens.lower().endswith("xsf") ):
-            rho_tip, lvec_tip, nDim_tip, head_tip = io.loadXSF( args.tip_dens )
+        #  No need to renormalize: fieldFFT already works with density
+        print(">>> Loading tip density from ", args.tip_dens, "...")
+        if args.tip_dens.lower().endswith("xsf"):
+            rho_tip, lvec_tip, n_dim_tip, head_tip = io.loadXSF(args.tip_dens)
         else:
-            raise ValueError('ERROR!!! Unknown or unsupported format of the tip density file "'+args.tip_dens+'"\n')
-        if bSubtractCore:
+            raise ValueError('ERROR!!! Unknown or unsupported format of the tip density file "' + args.tip_dens + '"\n')
```
[the remark this answers is not in the document]
Comment (yakutovicha) [reply ppafm/cli/generateElFF.py]: done in https://github.com/Probe-Particle/ppafm/pull/215/commits/07749da3fff952f2b0ad29af1db78bd24adabd30

File: ppafm/cli/generateElFF.py [hunk 7]
```diff
@@ -1,178 +1,184 @@
...
-        if (geometry_format=="xsf" and args.KPFM_tip.lower().endswith(".xsf")):
-            Vref_t = args.Vref
+            raise ValueError(
+                'ERROR!!! Format of the "'
+                + args.KPFM_sample
+                + '" file with Hartree potential under bias is unknown\n'
+                + ' or incompatible with the main input format, which is "'
+                + input_format
+                + ".\n"
+            )
```
[the remark this answers is not in the document]
Comment (yakutovicha) [reply ppafm/cli/generateElFF.py]: done in https://github.com/Probe-Particle/ppafm/pull/215/commits/07749da3fff952f2b0ad29af1db78bd24adabd30

File: ppafm/cli/plot_results.py [hunk 8]
```diff
@@ -62,413 +63,343 @@ def main():
...
+    print("Ks   =", k_constants)
+    print("Qs   =", charges)
+    print("Amps =", amplitudes)
     print(" ============= RUN  ")
 
-    dz = PPU.params["scanStep"][2]
-    xTips, yTips, zTips, lvecScan = PPU.prepareScanGrids()
-    extent = (xTips[0], xTips[-1], yTips[0], yTips[-1])
+    dz = common.params["scanStep"][2]
+    x_tips, y_tips, z_tips, _ = common.prepareScanGrids()
```
[the remark this answers is not in the document]
Comment (yakutovicha) [reply ppafm/cli/plot_results.py]: done in https://github.com/Probe-Particle/ppafm/pull/215/commits/d44f71263df3173a92aa843d74fa34150375b1e1

File: ppafm/cli/relaxed_scan.py (hunk 3 above)
Comment (yakutovicha) [reply ppafm/cli/relaxed_scan.py]: done in https://github.com/Probe-Particle/ppafm/pull/215/commits/8b7de26f991732f1e2a2b02cc6fdada6cef46ed9

File: ppafm/cli/relaxed_scan.py (hunk 2 above)
Comment (yakutovicha) [reply ppafm/cli/relaxed_scan.py:33]: done in https://github.com/Probe-Particle/ppafm/pull/215/commits/b1616f1d8ea33d0c1c2074bd5a9f4b56ecd00bf5

**Review by NikoOinonen — CHANGES_REQUESTED** [review]

File: ppafm/cli/generateElFF.py [hunk 9]
```diff
@@ -1,178 +1,184 @@
...
-                             +' or incompatible with the main input format, which is "'+geometry_format+'.\n')
-
-        dV_kpfm = (V_kpfm - V_v0_aux)
-
-        print(">>> Loading tip density under bias from ",args.KPFM_tip,"...")
-        if (geometry_format=="xsf" and args.KPFM_tip.lower().endswith(".xsf")):
-            Vref_t = args.Vref
+            print(
+                'ERROR!!! Format of the "{args.KPFM_sample}" file with Hartree potential under bias is unknown\n'
+                + ' or incompatible with the main input format, which is "{input_format}".\n',
```
[the remark this answers is not in the document]
Comment (yakutovicha) [reply ppafm/cli/generateElFF.py]: done in https://github.com/Probe-Particle/ppafm/pull/215/commits/56c4f0ab8b2e36464e6a3a0f475fe54b34c399a6

**Review by NikoOinonen — APPROVED** [review]
LGTM!

File: ppafm/cli/generateElFF.py (hunk 1 above)
Comment (ProkopHapala) [reply ppafm/cli/generateElFF.py:72]: this allows the user to specify tip Multipole as dictionary in `tip.py` file

it is hacky, but why not

**Review by ProkopHapala — APPROVED** [review]
I don't actually see what are the changes. 
Just the formatng? 
But I don't see obvious reason why not to approve it

File: ppafm/cli/generateElFF.py (hunk 1 above)
Comment (ProkopHapala) [reply ppafm/cli/generateElFF.py:72]: cannot we just pass it as string like 
```
"{'dz2':-0.1,'pz':0.05,'s':-0.05}"
```
and than exec it? 

something like this:
```
if isinstance( common.params["tip"], str):
      exec( 'tipMultipole='+common.params["tip"] )
      common.params["tip"] = tipMultipole

```

Comment (NikoOinonen) [reply ppafm/cli/generateElFF.py:72]: I made a new issue for it: #221.

**Pull request merged**